ada/tasks.py

```diff
@@ -15,10 +19,44 @@
     soft_time_limit=10,
     time_limit=15,
 )
-def submit_whatsappid_to_rapidpro(whatsappid):
+def start_prototype_survey_flow(whatsappid):
     if rapidpro and settings.ADA_PROTOTYPE_SURVEY_FLOW_ID:
         return rapidpro.create_flow_start(
             extra={},
             flow=settings.ADA_PROTOTYPE_SURVEY_FLOW_ID,
             urns=[f"whatsapp:{whatsappid.lstrip('+')}"],
         )
+
+
+@app.task(
+    autoretry_for=(RequestException, SoftTimeLimitExceeded, TembaHttpError),
+    retry_backoff=True,
+    max_retries=15,
+    acks_late=True,
+    soft_time_limit=10,
+    time_limit=15,
+)
+def start_topup_flow(whatsappid):
+    if rapidpro and settings.ADA_TOPUP_FLOW_ID:
+        return rapidpro.create_flow_start(
+            extra={},
+            flow=settings.ADA_TOPUP_FLOW_ID,
+            urns=[f"whatsapp:{whatsappid.lstrip('+')}"],
+        )
+
+
+@app.task(
+    autoretry_for=(RequestException, SoftTimeLimitExceeded, TembaHttpError),
```

Comment:
TembaHttpError is a rapidpro specific error, not possible here

Reply:
Removed